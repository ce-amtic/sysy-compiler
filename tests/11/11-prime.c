#include<stdio.h>
const int N=100005;
int read()
{
    int x;
    scanf("%d",&x);
    return x;
}
int main() 
{
    int n,m,k=0,p[N];
    int vis[N];
	n=read();
    m=read();
    int i=1;
    while(i<=n) 
    {
        vis[i]=0;
        i=i+1;
    }
    i=2;
    while(i<N)
	{
		if(!vis[i]) 
        {
            k=k+1; 
            p[k]=i;
        }
        int j=1;
        while(j<=k&&i*p[j]<N)
		{
			vis[i*p[j]]=1;
			if(i%p[j]==0)
                break;
            j=j+1;
		}
        i=i+1;
	}
	while(m)
	{
		int x=read();
		printf("%d\n",p[x]);
        m=m-1;
	}
}