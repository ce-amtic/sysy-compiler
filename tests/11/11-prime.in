100000 7
1
2
3
4
5
1145
1919