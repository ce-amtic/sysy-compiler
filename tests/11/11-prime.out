2
3
5
7
11
9239
16573
